File: apps/android/app/src/main/java/app/yula/android/data/models/Conversation.kt
package app.yula.android.data.models

data class Conversation(
    val id: String,
    val title: String,
    val summary: String? = null,
    val messageCount: Int = 0,
    val lastMessageAt: String? = null,
    val createdAt: String,
    val updatedAt: String
)

File: packages/types/src/conversation.ts
import type { Message } from './message'

export interface Conversation {
  id: string
  userId: string
  title: string
  summary?: string
  messageCount: number
  lastMessageAt: Date
  createdAt: Date
  updatedAt: Date
}

export interface ConversationWithMessages extends Conversation {
  messages: Message[]
}

export interface CreateConversation {
  title?: string
}
